airbyte-integrations/connectors/source-relational-db/src/main/java/io/airbyte/integrations/source/relationaldb/AbstractDbSource.java

```diff
@@ -135,36 +135,49 @@ public AirbyteCatalog discover(final JsonNode config) throws Exception {
-
-    final Map<String, TableInfo<CommonField<DataType>>> fullyQualifiedTableNameToInfo =
-        discoverWithoutSystemTables(database)
-            .stream()
-            .collect(Collectors.toMap(t -> String.format("%s.%s", t.getNameSpace(), t.getName()), Function
-                .identity()));
-
-    validateCursorFieldForIncrementalTables(fullyQualifiedTableNameToInfo, catalog);
-
-    final List<AutoCloseableIterator<AirbyteMessage>> incrementalIterators =
-        getIncrementalIterators(database, catalog, fullyQualifiedTableNameToInfo, stateManager, emittedAt);
-    final List<AutoCloseableIterator<AirbyteMessage>> fullRefreshIterators =
-        getFullRefreshIterators(database, catalog, fullyQualifiedTableNameToInfo, stateManager, emittedAt);
-    final List<AutoCloseableIterator<AirbyteMessage>> iteratorList = Stream
-        .of(incrementalIterators, fullRefreshIterators)
-        .flatMap(Collection::stream)
-        .collect(Collectors.toList());
+    throws Exception {
+    try {
+      final StateManager stateManager =
+          StateManagerFactory.createStateManager(getSupportedStateType(config), deserializeInitialState(state, config), catalog);
+      final Instant emittedAt = Instant.now();
 
-    return AutoCloseableIterators
-        .appendOnClose(AutoCloseableIterators.concatWithEagerClose(iteratorList), () -> {
-          LOGGER.info("Closing database connection pool.");
-          Exceptions.toRuntime(this::close);
-          LOGGER.info("Closed database connection pool.");
-        });
+      final Database database = createDatabaseInternal(config);
+
+      final Map<String, TableInfo<CommonField<DataType>>> fullyQualifiedTableNameToInfo =
+          discoverWithoutSystemTables(database)
+              .stream()
+              .collect(Collectors.toMap(t -> String.format("%s.%s", t.getNameSpace(), t.getName()), Function
+                  .identity()));
+
+      validateCursorFieldForIncrementalTables(fullyQualifiedTableNameToInfo, catalog);
+
+      final List<AutoCloseableIterator<AirbyteMessage>> incrementalIterators =
+          getIncrementalIterators(database, catalog, fullyQualifiedTableNameToInfo, stateManager, emittedAt);
+      final List<AutoCloseableIterator<AirbyteMessage>> fullRefreshIterators =
+          getFullRefreshIterators(database, catalog, fullyQualifiedTableNameToInfo, stateManager, emittedAt);
+      final List<AutoCloseableIterator<AirbyteMessage>> iteratorList = Stream
+          .of(incrementalIterators, fullRefreshIterators)
+          .flatMap(Collection::stream)
+          .collect(Collectors.toList());
+
+      return AutoCloseableIterators
+          .appendOnClose(AutoCloseableIterators.concatWithEagerClose(iteratorList), () -> {
+            LOGGER.info("Closing database connection pool.");
+            Exceptions.toRuntime(this::close);
+            LOGGER.info("Closed database connection pool.");
+          });
+    } catch (final Exception exception) {
```

Comment:
Is this logic of surfacing an `AirbyteTraceMessage` something that can be tested within `PostgresSourceTest.java`, `MysqlSourceTest.java, etc? It seems there's already a test within `PostgresSourceTest.java`